[NL] Add damping factor to global Newton.

endJunction opened the pull request:
This adds damping factor to the global newton to the constructor and the prj file parser.
Is it OK to have this one to have a default value 1.0? Or shell I add `<damping>1<\damping>` to all projects?

TomFischer commented on NumLib/ODESolver/NonlinearSolver.h:
```diff
@@ -89,9 +89,11 @@ class NonlinearSolver<NonlinearSolverTag::Newton> final
         GlobalLinearSolver& linear_solver,
-        const unsigned maxiter)
+        const unsigned maxiter,
+            double const damping = 1.0)
```
Indentation looks defect.

TomFischer commented on NumLib/ODESolver/NonlinearSolver.cpp:
```diff
@@ -321,10 +321,12 @@ createNonlinearSolver(GlobalLinearSolver& linear_solver,
     }
     if (type == "Newton")
     {
+        //! \ogs_file_param{prj__nonlinear_solvers__nonlinear_solver__damping}
```
Is there already a corresponding documentation?

chleh commented on NumLib/ODESolver/NonlinearSolver.h:
```diff
@@ -117,8 +119,7 @@ class NonlinearSolver<NonlinearSolverTag::Newton> final
 
-    double const _alpha =
-        1;  //!< Damping factor. \todo Add constructor parameter.
+    double const _damping;  //!< Damping factor \in (0, 1]
```
Should the range be chacked in the code?

endJunction commented:
Formatted, docu provided, added a check for the factor being positive, removed the condition <=1 because someone (@nagelt ) might want to overdo it a little ;)
